### DISCUSSION TITLE
Adding atom results in strange smiles

### DISCUSSION THREAD

**Discussion by tdudgeon** [rdkit/rdkit#7876] [Q&A] [answered]
I am modifying a molecule by adding an atom, creating a bond and then setting x,y,z coordinates of atom in the conformer.
e.g. something like this:
```
idx = mol.AddAtom(Chem.Atom(atomic_no))
mol.AddBond(idx, attached_atom_num)
mol.GetConformer(-1).SetAtomPosition(idx, Geometry.Point3D(x, y, z))
```
In doing this the atom to which the new atom is attached become chiral.

When looking at the molfile that is generated from this modified molecule all looks good.
[A71EV2A-x1775_A_201_1_A71EV2A-x0526+A+147+1_ligand_LIG.txt](https://github.com/user-attachments/files/17210513/A71EV2A-x1775_A_201_1_A71EV2A-x0526%2BA%2B147%2B1_ligand_LIG.txt)

But when written to SMILES I get `CCOc1ccc(C(=O)C(C)~S)cc1`. Note the squiggle bond on the attached atom. CDK thinks this is invalid SMILES.
I've tried adding `Chem.AssignStereochemistryFrom3D(mol)` but that doesn't help.

**Comment by ptosco** [comment, marked as answer]
@tdudgeon Hi Tim the last parameter of [`RWMol.AddBond`](https://rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.RWMol.AddBond) defaults to `Chem.BondType.UNSPECIFIED`, you need to specify the bond order (e.g., `Chem.BondType.SINGLE`)

**Comment by tdudgeon** [comment]
Thanks. That's fixed it.
